### ISSUE TITLE
ValueError: External step can only transition to external step or external step done.

### ISSUE THREAD

**Issue by pizzakid25** [custom-components/alexa_media_player#1118]
**Describe the bug**
When attempting to login using the login proxy method you receive a "500 Internal Server Error" message when you are redirected to _0.0.0.0:8123/auth/alexamedia/callback?flow_id=REDACTED 32 CHARS_ after signing in on the Amazon page. When you go back to the configuration screen open in the other tab, it still shows that you are not logged in.

**To Reproduce**
1. Go to add the integration.
2. Enter your username and password. Leave everything else what it is.
3. Click sign-in on the Amazon Alexa page that has been opened.
4. Complete the captcha.
5. See the error.

**Expected behavior**
After completing the captcha, you are signed in.

**Screenshots**

**System details**
 - Home-assistant (version): 2021.1.4
 - Hassio (Yes/No): Yes
 - alexa_media  (version from `const.py` or HA startup): 3.5.0
 - alexapy (version from `pip show alexapy` or HA startup): 1.22.0
 - Amazon [2FA is enabled](https://github.com/custom-components/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account) (y/n). We will not debug login issues if unanswered: No

**Logs**
```
Logger: aiohttp.server
Source: data_entry_flow.py:162
First occurred: 6:13:45 PM (1 occurrences)
Last logged: 6:13:45 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 118, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 56, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 18, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 72, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 127, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 129, in handle
    result = await result
  File "/config/custom_components/alexa_media/config_flow.py", line 954, in get
    await hass.config_entries.flow.async_configure(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 162, in async_configure
    raise ValueError(
ValueError: External step can only transition to external step or external step done.
```
**Additional context**
I was thinking that this might have something to do with the 2021.1.3 update.

**Comment by alandtse** [comment]
Just to check, did this work for you on a prior version or is this the first time you've tried the proxy method? I assume if you disable the proxy you can log in correct?

**Comment by pizzakid25** [comment]
This is the first time I've tried the proxy method. If I disable the proxy, I get this error:
```
Translation Error: The intl string context variable "email" was not provided to the string "{email} - alexa.{url}
{message}
{captcha_image}"

Error connecting; check network and retry
```
This is shown in the log:
```
Logger: alexapy.helpers
Source: /usr/local/lib/python3.8/site-packages/alexapy/helpers.py:143
First occurred: 7:13:04 PM (1 occurrences)
Last logged: 7:13:04 PM

alexalogin.login((<alexapy.alexalogin.AlexaLogin object at 0x67af5670>,), {'data': {'proxy': False, 'password': 'REDACTED 8 CHARS', 'captcha': '73agg6', 'securitycode': ''}}): A connection error occured: An exception of type KeyError occurred. Arguments: ('Referer',)
```

**Comment by alandtse** [comment]
What is the domain you're logging into? 

Also, can you enable advanced debugging and enable logging on [alexapy](https://github.com/custom-components/alexa_media_player/wiki/FAQ#how-do-i-enable-debug-logging-for-the-component) for the attempt? You will need to scrub you data before posting though.

Did any prior versions like 3.4.1 work for you or is this the first time you've used the component?

**Comment by pizzakid25** [comment]
I am logging into the amazon.com domain using a phone number as the email (mobile account). Prior versions did work. I think the integration stopped working around version 3.5.0.
```
2021-01-23 08:25:25 DEBUG (MainThread) [custom_components.alexa_media.config_flow] Creating new login
2021-01-23 08:25:25 DEBUG (MainThread) [custom_components.alexa_media.helpers] 7******305: Returning uuid {'uuid': 'REDACTED 32 CHARS', 'index': 0}
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexalogin] Login created for 7******305 - amazon.com
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexaproxy] Starting proxy at http://192.168.231.133:8123:48471
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/?config_flow_id=REDACTED 32 CHARS&callback_url=http://192.168.231.133:8123/auth/alexamedia/callback?flow_id=REDACTED 32 CHARS
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexalogin] Resetting Login for 7******305 - amazon.com
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexalogin] Attempting oauth login to https://www.amazon.com/ap/signin?openid.return_to=https://www.amazon.com/ap/maplanding&openid.assoc_handle=amzn_dp_project_dee_ios&openid.identity=http://specs.openid.net/auth/2.0/identifier_select&pageId=amzn_dp_project_dee_ios&accountStatusPolicy=P1&openid.claimed_id=http://specs.openid.net/auth/2.0/identifier_select&openid.mode=checkid_setup&openid.ns.oa2=http://www.amazon.com/ap/ext/oauth/2&openid.oa2.client_id=device:REDACTED 92 CHARS&openid.ns.pape=http://specs.openid.net/extensions/pape/1.0&openid.oa2.response_type=token&openid.ns=http://specs.openid.net/auth/2.0&openid.pape.max_auth_age=0&openid.oa2.scope=device_auth_access&language=en_US
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexaproxy] Starting auth at https://www.amazon.com/ap/signin?openid.return_to=https://www.amazon.com/ap/maplanding&openid.assoc_handle=amzn_dp_project_dee_ios&openid.identity=http://specs.openid.net/auth/2.0/identifier_select&pageId=amzn_dp_project_dee_ios&accountStatusPolicy=P1&openid.claimed_id=http://specs.openid.net/auth/2.0/identifier_select&openid.mode=checkid_setup&openid.ns.oa2=http://www.amazon.com/ap/ext/oauth/2&openid.oa2.client_id=device:REDACTED 92 CHARS&openid.ns.pape=http://specs.openid.net/extensions/pape/1.0&openid.oa2.response_type=token&openid.ns=http://specs.openid.net/auth/2.0&openid.pape.max_auth_age=0&openid.oa2.scope=device_auth_access&language=en_US for domain amazon.com for configflow REDACTED 32 CHARS with callback http://192.168.231.133:8123/auth/alexamedia/callback?flow_id=REDACTED 32 CHARS
2021-01-23 08:25:25 DEBUG (MainThread) [alexapy.alexalogin] Unable to generate OTP; 2FA app key not configured
2021-01-23 08:25:26 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/ap/uedata?ld&v=0.215163.0&id=REDACTED 20 CHARS&sw=1366&sh=768&vw=1359&vh=649&m=1&sc=REDACTED 20 CHARS&ue=2&bb=59&cf=102&be=106&fp=115&fcp=115&pc=313&tc=-621&na_=-621&ul_=REDACTED 14 CHARS&_ul=REDACTED 14 CHARS&rd_=REDACTED 14 CHARS&_rd=REDACTED 14 CHARS&fe_=-610&lk_=-610&_lk=-610&co_=-610&_co=-610&sc_=REDACTED 14 CHARS&rq_=-594&rs_=-30&_rs=-15&dl_=-22&di_=118&de_=121&_de=121&_dc=312&ld_=312&_ld=REDACTED 14 CHARS&ntd=-1&ty=0&rc=0&hob=1&hoe=3&ld=314&t=REDACTED 13 CHARS&ctb=1&rt=cf:5-1-3-1-2-3-0__ld:18-9-3-1-4-5-0&csmtags=aui|aui:aui_build_date:3.19.8-2020-12-30|fls-na-amazon-com|netInfo:wifi|adblk_no&viz=visible:2&pty=AuthenticationPortal&spty=MobileSignInApplication&pti=undefined&tid=REDACTED 20 CHARS&aftb=1
2021-01-23 08:25:26 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/favicon.ico
2021-01-23 08:25:27 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/ap/uedata?at&v=0.215163.0&id=REDACTED 20 CHARS&ctb=1&m=1&sc=REDACTED 20 CHARS&pc=2094&at=2094&t=REDACTED 13 CHARS&csmtags=csm-feature-touch-enabled:false&pty=AuthenticationPortal&spty=MobileSignInApplication&pti=undefined&tid=REDACTED 20 CHARS&aftb=1
2021-01-23 08:25:27 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/ap/uedata?at&v=0.215163.0&id=REDACTED 20 CHARS&ctb=1&m=1&sc=REDACTED 20 CHARS&pc=2104&at=2104&t=REDACTED 13 CHARS&pty=AuthenticationPortal&spty=MobileSignInApplication&pti=undefined&tid=REDACTED 20 CHARS&aftb=1&ui=2
2021-01-23 08:25:31 DEBUG (MainThread) [alexapy.alexaproxy] Post Request: http://192.168.231.133:48471/ap/signin/REDACTED 19 CHARS
2021-01-23 08:25:31 DEBUG (MainThread) [alexapy.alexalogin] Unable to generate OTP; 2FA app key not configured
2021-01-23 08:25:32 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/ap/uedata?ld&v=0.215163.0&id=REDACTED 20 CHARS&sw=1366&sh=768&vw=1359&vh=649&m=1&sc=REDACTED 20 CHARS&ue=2&bb=41&cf=88&be=98&fp=67&fcp=67&pc=584&tc=-728&na_=-728&ul_=-30&_ul=-29&rd_=REDACTED 14 CHARS&_rd=REDACTED 14 CHARS&fe_=-721&lk_=-721&_lk=-721&co_=-721&_co=-721&sc_=REDACTED 14 CHARS&rq_=-713&rs_=-39&_rs=-18&dl_=-23&di_=104&de_=105&_de=105&_dc=584&ld_=584&_ld=REDACTED 14 CHARS&ntd=-1&ty=0&rc=0&hob=1&hoe=2&ld=585&t=REDACTED 13 CHARS&ctb=1&rt=cf:15-7-3-1-3-8-1__ld:19-9-3-2-5-10-1&csmtags=aui|aui:aui_build_date:3.19.8-2020-12-30|fls-na-amazon-com|netInfo:wifi|adblk_no&viz=visible:1&pty=AuthenticationPortal&spty=MobileSignInApplication&pti=undefined&tid=REDACTED 20 CHARS&aftb=1
2021-01-23 08:25:33 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/ap/uedata?at&v=0.215163.0&id=REDACTED 20 CHARS&ctb=1&m=1&sc=REDACTED 20 CHARS&pc=1716&at=REDACTED 20 CHARS&pty=AuthenticationPortal&spty=MobileSignInApplication&pti=undefined&tid=REDACTED 20 CHARS&aftb=1&ui=3
2021-01-23 08:25:34 DEBUG (MainThread) [alexapy.alexaproxy] Get Request: http://192.168.231.133:48471/ap/uedata?at&v=0.215163.0&id=REDACTED 20 CHARS&ctb=1&m=1&sc=REDACTED 20 CHARS&pc=2298&at=2298&t=REDACTED 13 CHARS&csmtags=csm-feature-touch-enabled:false&pty=AuthenticationPortal&spty=MobileSignInApplication&pti=undefined&tid=REDACTED 20 CHARS&aftb=1
2021-01-23 08:25:45 DEBUG (MainThread) [alexapy.alexaproxy] Post Request: http://192.168.231.133:48471/ap/signin
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexaproxy] Proxy success for 7******305 - amazon.com. Redirecting to: http://192.168.231.133:8123/auth/alexamedia/callback?flow_id=REDACTED 32 CHARS
2021-01-23 08:25:46 DEBUG (MainThread) [custom_components.alexa_media.config_flow] Checking proxy response for 7******305 - amazon.com
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexaproxy] Stopping proxy at http://192.168.231.133:8123:48471
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] Attempting to register with amazon.com
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] auth response <ClientResponse(https://api.amazon.com/auth/register) [200 OK]>
<CIMultiDictProxy('Server': 'Server', 'Date': 'Sat, 23 Jan 2021 13:25:46 GMT', 'Content-Type': 'application/json;charset=UTF-8', 'Content-Length': '5418', 'Connection': 'keep-alive', 'x-amz-rid': 'REDACTED 20 CHARS', 'x-amzn-RequestId': 'REDACTED 36 CHARS', 'X-Amz-Date': 'Sat, 23 Jan 2021 13:25:46 GMT', 'Cache-Control': 'no-cache, no-store, must-revalidate', 'Pragma': 'no-cache', 'Vary': 'Content-Type,Accept-Encoding,X-Amzn-CDN-Cache,X-Amzn-AX-Treatment,User-Agent')>
with
{"requested_extensions": ["device_info", "customer_info"], "cookies": {"website_cookies": [{"Value": "REDACTED 420 CHARS", "Name": "frc"}, {"Value": "REDACTED 212 CHARS", "Name": "map-md"}], "domain": ".amazon.com"}, "registration_data": {"domain": "Device", "app_version": "2.2.223830.0", "device_type": "REDACTED 13 CHARS", "device_name": "%FIRST_NAME%'s%DUPE_STRATEGY_1ST%Alexa Media Player", "os_version": "11.4.1", "device_serial": "REDACTED 32 CHARS", "device_model": "iPhone", "app_name": "Alexa Media Player", "software_version": "1"}, "auth_data": {"access_token": "REDACTED 268 CHARS"}, "user_context_map": {"frc": "REDACTED 420 CHARS"}, "requested_token_type": ["bearer", "mac_dms", "website_cookies"]}
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] Successfully registered Rob's Alexa Media Player device with Amazon
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] New access token received which expires at 2021-01-23 09:25:46.508121 in 0:59:59.999891
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] 5 cookies successfully exchanged for refresh token for domain .amazon.com
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] Attempting to discover CSRF token
2021-01-23 08:25:46 DEBUG (MainThread) [alexapy.alexalogin] CSRF token found from /spa/index.html
2021-01-23 08:25:47 DEBUG (MainThread) [alexapy.alexalogin] GET:
https://alexa.amazon.com/api/bootstrap returned 200:OK with response <CIMultiDictProxy('Content-Type': 'application/json', 'Content-Length': '145', 'Connection': 'keep-alive', 'Server': 'Server', 'Date': 'Sat, 23 Jan 2021 13:25:47 GMT', 'x-amz-rid': 'REDACTED 20 CHARS', 'x-amzn-RequestId': 'REDACTED 36 CHARS', 'Content-Encoding': 'gzip', 'Vary': 'Content-Type,Accept-Encoding,X-Amzn-CDN-Cache,X-Amzn-AX-Treatment,User-Agent', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 REDACTED 32 CHARS.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'IAD66-C1', 'X-Amz-Cf-Id': 'REDACTED 56 CHARS')>
2021-01-23 08:25:47 DEBUG (MainThread) [alexapy.alexalogin] Not logged in due to email mismatch
2021-01-23 08:25:47 DEBUG (MainThread) [alexapy.alexalogin] Resetting Login for 7******305 - amazon.com
```

**Comment by pizzakid25** [comment]
Hmm. If I add an email address to my Amazon and try to sign in using that, it works. I added a note about this in the wiki.

**Comment by alandtse** [comment]
I guess you may not be able to undo the change, but if you go to https://alexa.amazon.com/api/bootstrap, what does it show in your `customerEmail`? I have a suspicion it has changed since the verification code hasn't changed.

**Comment by pizzakid25** [comment]
The `customerEmail` field shows my email. Before, I believe that it showed my mobile number.

**Comment by alandtse** [comment]
Yah. I'm wondering if it changed to something non blank. That would explain the error you got.

**Comment by pizzakid25** [comment]
I think I figured it out. Typically, if a user has both their email and mobile number linked to their Amazon account, they could use either to login to the regular Amazon website. But, for the integration, if you have an email linked to your account you must use your email to sign into the integration because the `customerEmail` field always returns the email if it is available. I think I worded my comment earlier a little funny. I added an email to my account shortly before I got the original error since I was configuring IFTTT to work with Alexa and it asked me to add an email. Thus, adding an email to my account changed the `customerEmail` field to show my email making it non blank. Due to how alexapy is configured, it tried to match my mobile number to my email which caused an email mismatch.

Only Email Linked: Use Email to Sign In
Only Mobile Number Linked: Use Mobile Number to Sign In
Mobile Number and Email Linked: Use Email to Sign In
